### ISSUE TITLE
Hint "x = true ===> x" is triggered by simple object creation or optional arguments

### ISSUE THREAD

**Issue by Grenkin1988** [fsprojects/FSharpLint#423]
### Description

Tool is reporting error `(FL0065): Hints `x = true` might be able to be refactored into `x`.` for the simple object creation process or optional arguments

### Repro steps

Optional argument:
1. `Http.AsyncRequest("someSite", silentHttpErrors = true) |> Async.RunSynchronously`
2. -> `(FL0065): Hints - Error: `x = true` might be able to be refactored into `x`. on the (48 - 48)`

Object initialization:
1. Given C# class:
```
public class MySimpleType
    {
        public int SimpleInt { get;set;}
        public bool SuperBoolProperty { get; set; }
        public string SimpleString { get;set;}
    }
```
2. And initialize it in F#
```
let giveObject () =
    MySimpleType(
        SimpleInt = 12,
        SuperBoolProperty = true,
        SimpleString = "String")
```
3. -> `(FL0065): Hints - Error: `x = true` might be able to be refactored into `x`. on the (20- 20)`

### Expected behavior

Should be no error

### Actual behavior

Tool reports error

**Comment by Grenkin1988** [comment]
Actually same happens for the Hint `"x = null ===> isNull x",`
Just modify:
```
let giveObject () =
    MySimpleType(
        SimpleInt = 12,
        SuperBoolProperty = true,
        SimpleString = null)
```

**Comment by Evangelink** [comment]
I am also facing the same issue for both `true` and `false` cases on property initialization in the ctor declaration.
